src/sqlfluff/dialects/dialect_mysql.py

```diff
@@ -165,6 +168,17 @@
             Ref("NumericLiteralSegment"),
         ),
     ),
+    QuotedLiteralSegment=AnyNumberOf(
+        # MySQL allows whitespace-concatenated string literals (#1488).
+        # Since these string literals can have comments between them,
+        # we use grammar to handle this.
+        NamedParser(
+            "single_quote",
```

Comment:
I read somewhere that MySQL also supports double-quoted strings. Are we handling this?

https://dev.mysql.com/doc/refman/8.0/en/string-literals.html

Reply:
I hadn't considered the double-quoted strings, let me take a look tomorrow and hopefully shouldn't be too hard to extend 👍

Reply:
@barrywhart So double-quoted strings are disabled if `ANSI_QUOTES` SQL mode is enabled.

When I use the MySQL docker image ([source](https://hub.docker.com/_/mysql/)) to test this, it looks like this is enabled by default:
```
docker run --platform linux/amd64 --name some-mysql -e MYSQL_ROOT_PASSWORD=my-secret-pw -d -p 3306:3306 mysql
```
![image](https://user-images.githubusercontent.com/80432516/150315017-9669e08b-f421-4935-a042-56ab4e3b487b.png)

Given that by default (at least in docker) double quotes are still for quoted identifiers, and we don't support flags for different options, I would suggest we stick with the default behaviour.

We're currently parsing the double quote as a string literal but perhaps, given the above, we should also remove this and parse as a quoted identifier.
![image](https://user-images.githubusercontent.com/80432516/150317931-25b46618-8224-4a28-87b5-06e1a9fe571f.png)

Reply:
I've posted on #general on slack in case anyone disagrees with my approach above: https://sqlfluff.slack.com/archives/C01GPHR8J0G/p1642673594037700

Reply:
Sounds good. Thanks for checking into this!